jekyll/_cci2/ios-migrating-from-1-2.md

```diff
@@ -0,0 +1,386 @@
+
+## Copying your build commands to the 2.0 config file
+
+### Job name and Xcode version
+
+In the 2.0 config file the first few lines specify the name of the job
+we will be running for you, and the Xcode version to use:
+
+```
+version: 2
+jobs:
+  build-and-test:
+    macos:
+      xcode:
+        version: "8.3.3"
+...
+  deploy:
+    macos:
+      xcode:
+        version: 8.3.3
+...
+```
+
+### Build steps key
+
+The top-level `steps` key contains all the build steps that will be run
+for a particular job:
+
+```
+jobs:
+  build-and-test:
+    steps:
+      - ...
+      - ...
+```
+
+You can see all the available step types in [this doc]({{ site.baseurl }}/2.0/configuration-reference/).
+Please bear in mind that Docker support is not available in the macOS builds.
+
+### Checking out the project code
+
+One of the first items under `steps` will be the code checkout step:
+
+```
+jobs:
+  build-and-test:
+    steps:
+      - checkout
+```
+
+### Caching Ruby gems installed via Bundler
+
+In CircleCI 2.0, cache save and cache restore are based on a _cache
+key_. Here is how you can cache the Ruby gems based on the content of
+`Gemfile.lock`:
+
+```
+jobs:
+  build-and-deploy:
+    steps:
+      ...
+      - restore_cache:
+          keys:
+          - v1-gems-{{ checksum "Gemfile.lock" }}
+          # Fall back to using the latest cache if no exact match is found.
+          - v1-gems-
+      # Install gems.
+      - run:
+          name: Bundle install
+          command: bundle check || bundle install
+          environment:
+            BUNDLE_JOBS: 4
+            BUNDLE_RETRY: 3
+            # This is the path where all the gems will be installed, and
+            # which we will later cache.
+            BUNDLE_PATH: vendor/bundle
+      - save_cache:
+          key: v1-gems-{{ checksum "Gemfile.lock" }}
+          paths:
+            - vendor/bundle
+```
+
+Every time your Gemfile.lock changes, a new cache will be created.
+Please check out [this doc]({{ site.baseurl }}/2.0/caching/) for
+more information about cache keys and other available key options
+beyond `checksum`.
+
+### Running tests
+
+We recommend using [Fastlane Scan](https://github.com/fastlane/fastlane/tree/master/scan)
+to run your tests:
+
+```
+jobs:
+  build-and-deploy:
+    steps:
+      ...
+      - run:
+          name: Run tests
+          command: bundle exec fastlane scan
+          environment:
+            SCAN_DEVICE: iPhone 6
+            SCAN_SCHEME: WebTests
+```
+
+You can replace the `bundle exec fastlane scan` command with your custom
+test command, and change the environment variables passed into it. If
+your test command spans multiple lines, you can include multiple
+commands in a single step:
+
+```
+jobs:
+  build-and-deploy:
+    steps:
+      ...
+      - run:
+          name: Run tests
+          command: |
+            make build
+            make test
+```
+
+### Storing artifacts, test results and diagnostic files
```

Comment:
Storing artifacts, test results and diagnostic files -> Storing Artifacts, Test Results, and Diagnostic Files